module/solutions/helper.js

```diff
@@ -145,15 +159,25 @@ module.exports = class SolutionsHelper {
         }
 
         // adding solution id to the program components key
+
         if (solutionData.programExternalId) {
-          let updateProgram = await programsQueries.findOneAndUpdate(
-            {
-              _id: solutionData.programId,
-            },
-            {
-              $addToSet: { components: solutionCreation._id },
+          if (isExternalProgram) {
+            programData[0].components.push(solutionCreation._id.toString());
+            const updateProgram =
+                await projectService.programUpdate(userToken, solutionData.programId, {
```

Comment:
@praveenKDass, we should not update the project service program from here. Whoever is calling this API will update it in their own way.